weave/tests/test_basket.py

```diff
@@ -701,7 +702,11 @@ def test_basket_time_is_utc(test_pantry):
     match_iso8601 = re.compile(regex).match
     assert match_iso8601(str(manifest_dict['upload_time'])) is not None
 
-
+# Skip tests if pymongo is not installed.
+@pytest.mark.skipif(
+    "pymongo" not in sys.modules or not os.environ.get("MONGODB_HOST", False),
+    reason="Pymongo required for this test",
+)
```

Comment:
Why is pymongo required for this test?

Reply:
we should make Mongo is optional?

Reply:
I do agree with code not within the scope of pytest. *We should development a mechanism to determine if mongodb is available. That will take some time.

For codes in a pytest environment. Pytest already has this mechanism built-in, only the test cases that uses pymongo directly will need the mechanism.

For test case that uses pymongo indirectly, such as pantry.upload_basket. Our *developed mechanism should handle this case.

Reply:
Another thing that came to mind was that if pymongo is installed, that doesn't guarantee that mongodb is available.